src/main/java/org/commcare/xml/EndpointParser.java

```diff
@@ -0,0 +1,52 @@
+package org.commcare.xml;
+
+import org.commcare.suite.model.Endpoint;
+import org.commcare.suite.model.StackOperation;
+import org.javarosa.xml.ElementParser;
+import org.javarosa.xml.util.InvalidStructureException;
+import org.javarosa.xml.util.UnfullfilledRequirementsException;
+import org.kxml2.io.KXmlParser;
+import org.xmlpull.v1.XmlPullParserException;
+
+import java.io.IOException;
+import java.util.Vector;
+
+public class EndpointParser extends ElementParser<Endpoint> {
+
+    static final String NAME_ENDPOINT = "endpoint";
+    private static final String NAME_ARGUMENT = "argument";
+    private static final String NAME_STACK = "stack";
```

Comment:
Suggestion: Moving this ID to the StackOpsParser and re-using the static String between this and the other stack ops parsing contexts.